### DISCUSSION TITLE
What is the goal of SingleFileZ compared to SingleFile?

### DISCUSSION THREAD

**Discussion by user0022** [gildas-lormeau/SingleFileZ#91] [Q&A] [answered]
Hello! Excuse-me stupid question maybe!!

Well I am an average user, not very talented but with a little technical interest in computing as I use Linux and have learned some bases of html and css (very few but well...), which not everybody does.
Despite of it, I use the SingleFileZ addon and am pleased with it, but I am not sure at all of the goal of the zip transformation. What I see is that it takes less place on the disk, so as I like this feature I use SFZ. But is it made for something else? Security? Easier upload somewhere else? Reliability? Easier working on the file with some software?... 
The comment by Vik in https://addons.mozilla.org/fr/firefox/addon/singlefilez/reviews/?src=search saying "to save space" seems to indicate that the space is the advantage.

Maybe if it is the case, some users like me or with yet less knowledge, could happen to come accross the SingleFileZ addon page, and not install it, but if they had understood that it saves space on their disk, they would have. So maybe (suggestion) it could be a good thing to explain this interest of the addon compared to SingleFile, as it is just said that the page is zipped, and maybe not everybody knows what is the result of "zipping"...

**Comment by gildas-lormeau** [comment, marked as answer]
The 2 extensions are separated mostly for practical reasons. When I "discovered" that an extension like SingleFile could produce HTML/ZIP files, I was tempted to add this feature in SingleFile itself, as an option. However, I quickly noticed it was a bit risky because the changes to apply in the code to support the legacy format and this new format would have been too large. So, I decided to fork SingleFile in order to deliver faster this feature without taking too much risks.

I think the zip format offers 2 advantages:
 - files are smaller
 - files can be unzipped in order to extract resources like images or fonts more easily

I wished I could implement the encryption to protect pages with a password but the last time I tried, I was not able to produce valid zip files. I guess this would give another advantage. 

Note that SingleFileZ appears in the comparison table here: https://github.com/gildas-lormeau/SingleFile#singlefilez and the first question in the FAQ also explains the pros and cons: https://github.com/gildas-lormeau/SingleFileZ/blob/master/faq.md.

**Comment by user0022** [comment]
Thank you for clear answer! 

Nevertheless I still tell myself that maybe the advantage of page size reducing could me more explicitly indicated, as one can also imagine someone looking for this feature typing e. g. "reduce size" or "save space" or "smaller" or "lighten" in the Firefox addon search engine (or even in "a big general search engine"), which won't lead to this addon. But not thinking about typing "zip" which seems the only word in the page indicating this feature. (Previously there were tags in Firefox addon pages that could be used for this but it doesn't exist anymore.)

I should have clicked on the links that are in the addon page, the day I installed it! But too much in a hurry. It would have avoided an useless question but well it gave me also the opportunity to give my opinion about the other thing which is maybe not very useful too but well to me it seems to make sense. Now I watched them and it's instructive and answers my question yes.

**Comment by lumnikemel** [comment]
Do you plan to add the annotation features in SingleFile to SingleFileZ?

**Comment by gildas-lormeau** [comment]
Sorry, I forgot to answer to your question. Yes, I would like the add the annotation editor in SingleFileZ too.

**Reply by nomatica** [reply]
Just wanted to say that both these extensions rock!  Thank you for making them

**Comment by gildas-lormeau** [comment]
For the record, SingleFileZ has been merged into SingleFile.

**Comment by user0022** [comment]
Thank you for the information! (recieved by mail).
I still used SFZ waiting for the merging you had talked about, but as
I had uninstalled SF I didn't know you had already done it. So now I
just reinstalled SF and will uninstall SFZ.

**Reply by gildas-lormeau** [reply]
You're welcome! Don't forget to select the option "File format > format > self-extracting ZIP (universal)" in SingleFile.

**Comment by MarineWonder** [comment]
I came upon these extensions today and was pleasantly surprised.

I had been using SingleFile first, but its saved file sizes were too large, I knew there should be a way to compress these files and still keep them in a one-click format to view.    I had set-up SingleFile to download as HTML and prompt me for a filename.  Then i saw you had SingleFileZ and tried that.  I was pleasently surprised at the big improvement in file size reduction.  But for the life of me, I can't find a way for SingleFileZ file to save with any extension.  When I manually add the .HTM extension, my File Explorer can then open the file as an HTM without a fuss.  I then came here and saw you had incorporated SingleFileZ into SingleFile, so hoping the save file extension feature was maybe not working, or somewhere hidden in options, but more fully developed on SingleFile, I removed SingleFileZ and went back and enabled the file format as self-extracting ZIP option in SingleFile, but while stillon the same page I've been testing this on, am now getting a SingleFile error: e.runtime.getContexts is not a function error. 

Currently both extensions are enabled and on the particular website I want to save, SingleFile extension icon shows ERR, and SingleFileZ icon shows OK.

So in hopes having had both extensions together on Chrome was causing the error, I removed both and started Chrome again.

I only installed SingleFile, left everything as defaults but still gets that error.

I uninstalled and installed SingleFileZ and at least it saves the page, I just have to rename it manually and give it an HTM extension.

I feel I'm doing something wrong in the extension options, but I can't figure it out.

Thanks again for the extensions!

**Reply by gildas-lormeau** [reply]
That's a strange issue, what version of Chrome are you using?
